src/realizations/catchment/Bmi_Multi_Formulation.cpp

```diff
@@ -186,18 +186,11 @@ const bool &Bmi_Multi_Formulation::get_allow_model_exceed_end_time() const {
  * @return The collection of forcing output property names this instance can provide.
  * @see ForcingProvider
  */
-boost::span<const std::string> Bmi_Multi_Formulation::get_available_variable_names() {
-    if (is_model_initialized() && available_forcings.empty()) {
-        for (const nested_module_ptr &module: modules) {
-            for (const std::string &out_var_name: module->get_bmi_output_variables()) {
-                available_forcings.push_back(module->get_config_mapped_variable_name(out_var_name));
-            }
-        }
-    }
+boost::span<const std::string> Bmi_Multi_Formulation::get_available_variable_names() const {
     return available_forcings;
```

Comment:
Where does this get initialized in the `Bmi_Multi_Formulation` case? I don't think it shares that code with `Bmi_Module_Formulation`

Reply:
Hmm I think for each model in the multi formulation, when `init_nested_module<T>` is called, it looks like it calls

```
Bmi_Module_Formulation::create_formulation ->
  Bmi_Module_Formulation::inner_create_formulation ->
  if (model_initialized) { ... available_forcings.push_back ... }
```

so, eventually they are initialized via `Bmi_Module_Formulation` -- but I'm not sure that it's correctly initialized, since I don't see any logic to handle duplicate names prior to `push_back`. I assumed the tests would check, but looking over them now, I don't see a call to `available_forcings` or `get_available_variable_names()`.

Reply:
That's initializing the `available_forcings` member of the individual modules, though. Without the deleted code here, where would the `Bmi_Multi_Formulation`'s member `available_forcings` get initialized as the concatenation of those?

(ignoring the fact that it should be checked as the union of disjoint sets - that's maybe an issue to file, but out of scope for this PR)

Reply:
Ohhh right 🤦🏾 I see now. I'll add back in the deleted code and move it within `create_multi_formulation` and that should handle it correctly then.

Reply:
I think I want to challenge you to add a test that exercises this. I think it never comes up currently because it would require something weird like a nested `Bmi_Multi_Formulation`, I guess inside another `Bmi_Multi_Formulation`. Otherwise, there's perhaps nothing that will actually call `Bmi_Multi_Formulation::get_available_variable_names()`

Reply:
I think there's maybe also a case where the realization configuration doesn't specify any output variables, but I'd need to read the code more carefully to be sure of that one hitting this.

If you want to investigate this somewhat dynamically, just add a `throw` in here, and see if anything ever runs into it.

Reply:
I'm not sure if it's what you had in mind, but if you get a chance to take a look, I added a simple test to ensure `get_available_variable_names()` returns expected values after construction of a Multi-BMI formulation. With the code deleted, as expected, the available forcings vector isn't filled.